Feature/oak detection

piasekj opened the pull request.

piasekj committed:
Yolo detection node

Can be run with pure ros command without external environments

piasekj committed:
Rosdep fix

Included necessary packages for dependency management using rosdep
Cleaned up setup.py

piasekj committed:
Bounding boxes topic

piasekj committed:
Change yolo node to publish on single topic

piasekj committed:
Aruco detection

Working on 2 different topics for now

piasekj committed:
Put Yolo and aruco detections on single topic

piasekj committed:
Aruco pose estimation

piasekj committed:
Change aruco marker size

piasekj committed:
Fix setup.py

piasekj committed:
Image publication encoding fix

piasekj committed:
Check model.device

FWalkowicz commented on src/trailblazer_detections/trailblazer_detections/yolo_node.py:
```diff
@@ -0,0 +1,185 @@
+
+        # Pose estimation parameters
+        # Camera calibration will be filled from CameraInfo
+        self.marker_length = 0.025  # Specified ArUco tags are 2.5 cm across
```
czy to zadziałą poprawnie dla naszego przypadku? W wytycznych markery ARUCO mają wymiary 20x20cm czyli `self.marker_length = 0.2`.

FWalkowicz reviewed (changes requested):
Czy można by dodać obliczanie odległości do wykrytych obiektów?

FWalkowicz commented:
Jeżeli chodzi o obliczanie odległości do obiektów:
1. Markery ARUCO, zobaczyłbym na to `cv2.aruco.estimatePoseSingleMarkers`
2. przedmioty z YOLO, to opcją jest obliczanie na podstawie PC2 z kamery ale nw jak to będzie obliczeniowo kosztowało

bafaurazan commented:
there is no README of launching detection

piasekj committed:
Aruco marker distance estimation

piasekj committed:
Custom msgs for detection

Msgs for trailblazer_detections implemented in trailblazer_interfaces

piasekj committed:
Add README

Also create PC2 distance estimation for YOLO model

piasekj committed:
trailblazer_detections bringup

Located in: TrailblazerML/src/trailblazer_bringup/oak_detection.launch.py

Launches:
- ros2 launch depthai_ros_driver camera.launch.py
- ros2 launch trailblazer_detections yolo_node.launch.py
